api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/core/Permute.kt

```diff
@@ -0,0 +1,55 @@
+package org.jetbrains.kotlinx.dl.api.core.layer.core
+
+import org.jetbrains.kotlinx.dl.api.core.KGraph
+import org.jetbrains.kotlinx.dl.api.core.layer.Layer
+import org.jetbrains.kotlinx.dl.api.core.shape.TensorShape
+import org.tensorflow.Operand
+import org.tensorflow.Shape
+import org.tensorflow.op.Ops
+
+/**
+ * Permutes the dimensions of the input according to a given pattern.
+ * Input shape: Arbitrary
+ * Output shape: Same as input shape, but with the dimensions re-ordered according to the specified pattern.
+ * @property [dims] array of integers.
+ * @property [name] Custom layer name.
+ * @constructor Creates [Permute] object.
+ */
+public class Permute(
+    public val dims: IntArray,
+    name: String
+) : Layer(name) {
```

Comment:
It would be a good thing to check if the values in `dims` are actually valid (similar to Keras implementation):
```kotlin
init {
    require(dims.sorted() == (1..dims.size).toList()) {
        "The values in dims should be consecutive and start from 1."
    }
}
```